### PR TITLE
AMReX: Update to 23.06

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: A/AMReX/build_tarballs.jl [hunk 1]
```diff
@@ -79,9 +79,16 @@ products = [
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = supported_platforms()
+platforms = expand_cxxstring_abis(platforms)
+
+# Windows (and only Windows) platforms somehow depend on libgfortran
+platforms = [
+    filter(p -> os(p) ≠ "windows", platforms);
```
Comment (giordano) [inline A/AMReX/build_tarballs.jl:86]:
```suggestion
    filter(!Sys.iswindows, platforms);
```

File: A/AMReX/build_tarballs.jl [hunk 2]
```diff
@@ -79,9 +79,16 @@ products = [
...
+    expand_gfortran_versions(filter(p -> os(p) == "windows", platforms));
```
Comment (giordano) [inline A/AMReX/build_tarballs.jl:87]:
```suggestion
    expand_gfortran_versions(filter(Sys.iswindows, platforms));
```

**Comment by eschnett** [comment]
@giordano The build fails on Windows because GCC 6 is too old, it does not support C++17. I say `preferred_gcc_version = v"9"`, but this does not help. The log output has:
```
_bk;t=1685826865579[0m[1m[21:14:25] [22m-- The C compiler identification is GNU 6.1.0
_bk;t=1685826865613[0m[1m[21:14:25] [22m-- The CXX compiler identification is GNU 6.1.0
```

**Comment by giordano** [comment]
libgfortran3 corresponds to gcc 4-6, libgfortran4 is only gcc 7, libgfortran5 is gcc 8+.  So no, there's no gcc 9 for libgfortran3 and 4 (and this is also why the option is called "preferred", you ask for what you'd like to use, but the actual compiler provided may be something else, depending on [lots of constraints](https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/8ce1e2700dae921aff69cd12b9aef4429913e90c/src/Rootfs.jl#L413-L511))

File: A/AMReX/build_tarballs.jl (hunk 1 above)
Comment (eschnett) [reply A/AMReX/build_tarballs.jl]: Done.

**Pull request merged**